repo/docs/Single_Sign_On.md

```diff
@@ -0,0 +1,87 @@
+# Single Sign On (SSO) Integration
+
+- [OIDC](#oidc) 
+- [SAML](#saml)
+
+## Introduction
+
+SSO is supported via [OIDC](https://openid.net/developers/how-connect-works/) or [SAML](https://en.wikipedia.org/wiki/Security_Assertion_Markup_Language).
+
+## <a name="oidc"></a>OIDC
+
+### Configuration
+
+Supply a map as below.
+
+```yaml
+global:
+  conf:
+    sso:
+      enabled: true
+      protocol: "oidc"
+      oidc:
+        clients:
+          arkcase:
+            registrationId: "id"
+            clientId: "******"
+            clientSecret: "******"
+            redirectUri: "https://localhost:8443/arkcase/login/oauth2/code/cognito"
+            authorizationUri: "...."
+            tokenUri: "...."
+            jwkSetUri: "...."
+            usernameAttribute: "email"
+            userInfoUri: "...."
+            scope: "email,openid"
+            responseType: "code"
+            responseMode: "form_post"
+            usersDirectory: "...."
+```
+
+The `authorizationUri`, `tokenUri`, `jwkSetUri`, and `userInfoUri` come from your OIDC provider's well known metadata URL. Your identity provider administrator will provide you this URL.
+
+For classic ArkCase (which uses an AngularJS UI, and has a version below 2024), the `registrationId` must match the last path element of the `redirectUri`.  The `redirectUri` must be of the form: `https://$baseUrl/arkcase/login/oauth2/code/$registrationId`.
+
+This document will be updated when OIDC is supported on modern ArkCase (Angular UI, versions 2024 and above).
+
+The `scope`, `responseType`, and `responseMode` shown above work in most situations, but can be changed as circumstances demand for particular OIDC services or customer requirements.
+
+`usersDirectory` must be the LDAP Spring bean name of the directory holding the user records for the OIDC users.  By convention, this bean name is formed by replacing the dots in the domain name with underscores.  So, for a login domain `dev.arkcase.com`, the `usersDirectory` is `dev_arkcase_com`.
```

Comment:
This can definitely be computed. We have an LDAP-related framework in place which can easily help us elucidate this value.